sql/expression/comparison.go

```diff
@@ -224,14 +228,30 @@ func (re *Regexp) compareRegexp(ctx *sql.Context, row sql.Row) (interface{}, err
 		return nil, err
 	}
 
-	if re.r == nil {
-		re.r, err = regex.New(regex.Default(), right.(string))
-		if err != nil {
-			return false, err
+	if re.pool == nil {
+		re.pool = &sync.Pool{
+			New: func() interface{} {
+				r, e := regex.New(regex.Default(), right.(string))
+				if e != nil {
+					err = e
+					return nil
+				}
+				return r
+			},
 		}
 	}
 
-	return re.r.Match(left.(string)), nil
+	r := re.pool.Get()
+	if r == nil {
+		return nil, err
+	}
+
+	var b bool
+	if matcher, ok := r.(regex.Matcher); ok {
```

Comment:
I would just do `r.(regex.Matcher).Match(left.(string))` and assume it's a `regex.Matcher`, it may be an `interface{}` but we're certain of what's inside.